### PR TITLE
Feature/dashboard filters pane

### PR DESCRIPTION
**Pull request by ShrawanBK**

### CODE REVIEW

File: src/components/ChoroplethLegend/styles.css [hunk 1]
```diff
@@ -3,29 +3,32 @@
     min-height: 100px;
 
     .heading {
-        margin-bottom: 0;
+        margin-bottom: var(--spacing-medium);
+        text-transform: uppercase;
+        color: #8196c2;
```
Comment (AdityaKhatri) [inline src/components/ChoroplethLegend/styles.css]: Define this color in app/styles.css

File: src/components/Navbar/styles.css [hunk 2]
```diff
@@ -1,42 +1,42 @@
...
+    align-items: center;
+    background-color: var(--color-backdrop-light);
+    padding: 0 calc(var(--spacing-large) - var(--spacing-medium));
     color: var(--color-text-on-primary);
 
     .app-brand {
         flex-shrink: 0;
-        padding: var(--spacing-small) var(--spacing-medium);
-        font-size: var(--font-size-large);
+        /* padding: var(--spacing-small) var(--spacing-medium); */
```
Comment (AdityaKhatri) [inline src/components/Navbar/styles.css]: Remove this instead of commenting

File: src/components/ProgramSelectors/SelectorItem/styles.css [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    .centered {
+        justify-content: center;
+    }
+    
+    .tree-input {
+        background-color: #eeeeee;
+    }
+}
+
```
Comment (AdityaKhatri) [inline src/components/ProgramSelectors/SelectorItem/styles.css]: Use consistent variables here
Do not worry about making it look similar to mockup.
Lets make it consistent first and we can ask the designer to make it consistent.

File: src/components/SegmentInput/Option/styles.css [hunk 4]
```diff
@@ -26,7 +22,8 @@
     }
 
     &.active {
-        background-color: var(--color-input-background);
+        border-radius: calc(4 * var(--radius-input));
+        background-color: #FFC0CBaa;
```
Comment (AdityaKhatri) [inline src/components/SegmentInput/Option/styles.css]: Define this in app/styles.css

File: src/views/Dashboard/ExploreData/styles.css [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    &:hover {
+        cursor: default;
+
+        .actions {
+            display: flex;
+            flex-direction: column;
+            visibility: visible;
+            background: var(--color-foreground);
+            padding: var(--spacing-small) 0;
+            width: calc(7 * var(--spacing-extra-large));
```
Comment (AdityaKhatri) [inline src/views/Dashboard/ExploreData/styles.css]: Use static value instead of this calculation

File: src/views/Dashboard/index.tsx [hunk 6]
```diff
@@ -730,36 +733,139 @@ const Dashboard = (props: Props) => {
...
+                        pending={covidFieldsPending}
+                    />
+                    {selectedFiveWOption && !isFiveWOptionKey(selectedFiveWOption) && (
+                        <SelectInput
+                            label="DFID Data Options"
+                            value={selectedFiveWSubOption}
+                            onChange={setFiveWSubOption}
+                            className={styles.inputItem}
+                            optionKeySelector={item => item}
+                            optionLabelSelector={item => item}
```
Comment (AdityaKhatri) [inline src/views/Dashboard/index.tsx]: Define this separately

File: src/views/Dashboard/index.tsx [hunk 7]
```diff
@@ -730,36 +733,139 @@ const Dashboard = (props: Props) => {
...
+                    />
+                </div>
+            </div>
+            <div className={styles.summary}>
+                <div
+                    className={styles.title}
+                >
+                    Summary
+                </div>
+                <Summary />
```
Comment (AdityaKhatri) [inline src/views/Dashboard/index.tsx]: If summary is used only here, lets move the title inside the component as well

File: src/views/Dashboard/styles.css [hunk 8]
```diff
@@ -19,7 +19,9 @@
 .dashboard {
     display: flex;
     position: relative;
+    align-self: center;
     flex-direction: column;
+    width: 87%;
```
Comment (AdityaKhatri) [inline src/views/Dashboard/styles.css]: Make it 100%

File: src/components/ProgramSelectors/SelectorItem/index.tsx [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+        collapseLevel = 0,
+        expandedFilters,
+        setExpandedFilters,
+        isMinimized,
+        icon,
+        searchText,
+        setSearchText,
+    } = props;
+
+    const isFilterExpanded = !!expandedFilters.find(e => e === name);
```
Comment (AdityaKhatri) [inline src/components/ProgramSelectors/SelectorItem/index.tsx]: Use useMemo

File: src/components/ChoroplethLegend/styles.css [hunk 10]
```diff
@@ -3,29 +3,32 @@
...
-            display: flex;
+            /* display: flex; */
             flex-grow: 1;
 
             .color {
                 flex-shrink: 0;
-                width: var(--font-size-medium);
-                height: var(--font-size-large);
+                width: calc(6 * var(--spacing-medium));
+                height: var(--spacing-large-medium);
```
Comment (tnagorra) [inline src/components/ChoroplethLegend/styles.css]: We shouldn't use "spacing" to define width/height. Instead create variables for them or just define them here

File: src/components/PrintButton/index.tsx [hunk 11]
```diff
@@ -70,9 +70,7 @@ function PrintButton(props: Props) {
                     onClick={handlePrintPreviewClick}
                     icons={<AiOutlinePrinter />}
                     disabled={disabled}
-                >
-                    Print preview
-                </Button>
+                />
```
Comment (tnagorra) [inline src/components/PrintButton/index.tsx:74]: Add a "title" prop if there is not content or the content is icon

File: src/views/Dashboard/styles.css [hunk 12]
```diff
@@ -155,7 +268,9 @@
         height: 210mm!important;
 
         .print-mode-button {
-            top: calc(var(--height-print-bar) + var(--spacing-medium));
+            /* top: calc(var(--height-print-bar) + var(--spacing-medium)); */
```
Comment (tnagorra) [inline src/views/Dashboard/styles.css]: Remove comment

File: src/views/Dashboard/styles.css [hunk 13]
```diff
@@ -83,11 +187,11 @@
             align-self: flex-start;
         }
 
-        .separator {
+        /* .separator {
             margin: var(--spacing-large) calc(-1 * var(--spacing-large));
             border-bottom: var(--width-separator-thin) solid var(--color-separator-light);
             width: calc(100% + var(--spacing-large) * 2);
-        }
+        } */
```
Comment (tnagorra) [inline src/views/Dashboard/styles.css]: Remove comment

File: src/views/Dashboard/styles.css [hunk 14]
```diff
@@ -100,9 +204,9 @@
             color: var(--color-accent);
         }
 
-        .input-item {
+        /* .input-item {
             margin: var(--spacing-medium) 0;
-        }
+        } */
```
Comment (tnagorra) [inline src/views/Dashboard/styles.css]: Remove comment

File: src/views/Dashboard/styles.css [hunk 15]
```diff
@@ -42,19 +44,117 @@
...
         flex-direction: column;
         transition: .2s left ease-in-out;
-        border: var(--width-separator-thin) solid var(--color-separator);
-        background-color: var(--color-foreground);
+        /* border: var(--width-separator-thin) solid var(--color-separator); */
+        background-color: var(--color-background);
         width: var(--width-config-container);
-        max-height: calc(100% - 3 * var(--spacing-medium) - 1em - 3 * var(--spacing-small));
+        height: 100%;
+        /* max-height: calc(100% - 3 * var(--spacing-medium) - 1em - 3 * var(--spacing-small)); */
```
Comment (tnagorra) [inline src/views/Dashboard/styles.css]: Remove comment

File: src/views/Dashboard/index.tsx [hunk 16]
```diff
@@ -730,36 +733,139 @@ const Dashboard = (props: Props) => {
...
-                        className={styles.regionSelector}
-                        onRegionLevelChange={setRegionLevel}
-                        regionLevel={regionLevel}
-                        searchHidden
-                    />
+                <FiltersPanel
+                    className={styles.filtersPanel}
+                    isMinimized={isFilterMinimized}
+                />
+                {/* <div className={styles.filters}>
```
Comment (tnagorra) [inline src/views/Dashboard/index.tsx]: Just remove everything. We don' need any of this.

File: src/components/ProgramSelectors/SelectorItem/index.tsx [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+                ) : (
+                    <IoMdArrowDropdown
+                        onClick={onExpandFilter}
+                        className={styles.collapseIcon}
+                    />
+                )}
+                <div className={styles.icon}>
+                    {icon}
+                </div>
+                {/* <IoIosDocument className={styles.icon} /> */}
```
Comment (tnagorra) [inline src/components/ProgramSelectors/SelectorItem/index.tsx]: Remove this comment

File: src/views/Dashboard/Summary/index.tsx [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+import styles from './styles.css';
+import SummaryItem from './SummaryItem';
+
+interface Summary {
+    allocatedBudget: number;
+    program: number;
+    partner: number;
+    component: number;
+    sector: number;
+}
```
Comment (tnagorra) [inline src/views/Dashboard/Summary/index.tsx]: This doesn't look correct! Is this used anywhere?

**Review by frozenhelium — APPROVED** [review]

**Pull request merged**